pkg/ddevapp/amplitude_project.go

```diff
@@ -0,0 +1,134 @@
+package ddevapp
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ddev/ddev/pkg/amplitude"
+	"github.com/ddev/ddev/pkg/dockerutil"
+	"github.com/ddev/ddev/pkg/globalconfig"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
+	"github.com/ddev/ddev/third_party/ampli"
+	"github.com/denisbrodbeck/machineid"
+)
+
+// ProtectedID returns the unique hash value for the project.
+func (app *DdevApp) ProtectedID() string {
+	appID, _ := machineid.ProtectedID("ddev" + app.Name)
+	return appID
+}
+
+// TrackProject collects and tracks information about the project for instrumentation.
+func (app *DdevApp) TrackProject() {
+	runTime := util.TimeTrack(time.Now(), "TrackProject")
+	defer runTime()
+
+	// Initialization is currently done before via init() func somewhere while
+	// creating the ddevapp. This should be cleaned up.
+	// TODO remove once clean up has done.
+	amplitude.InitAmplitude()
+
+	// Early exit if instrumentation is disabled.
+	if ampli.Instance.Disabled {
+		return
+	}
+
+	containersOmitted := app.GetOmittedContainers()
+
+	var services []string
+	containers, err := dockerutil.GetAppContainers(app.Name)
+	if err == nil {
+		for _, k := range containers {
+			serviceName := strings.TrimPrefix(k.Names[0], "/")
+			shortName := strings.Replace(serviceName, fmt.Sprintf("ddev-%s-", app.Name), "", 1)
+			services = append(services, shortName)
+		}
+	}
+
+	builder := ampli.Project.Builder().
+		Containers(services).
+		ContainersOmitted(containersOmitted).
+		DocumentRoot(app.GetDocroot()).
```

Comment:
Not sure if this has any value for instrumentation @rfay

Reply:
I agree. I've never looked at it. I guess the only reason it could exist is if we discover people using new and exotic docroot values. If we were debugging with this we might want it, but we really don't use it.

Reply:
I don't think we need values here for debugging purposes! Will remove it...